assessment/collectors/app_migration_analysis.go

```diff
@@ -103,7 +103,7 @@ func NewMigrationSummarizer(ctx context.Context, googleGenerativeAIAPIKey *strin
 		modelPro:                      client.GenerativeModel("gemini-1.5-pro-002"),
 		modelFlash:                    client.GenerativeModel("gemini-2.0-flash-001"),
 		conceptExampleDB:              conceptExampleDB,
-		dependencyAnalyzer:            dependencyAnalyzer.AnalyzerFactory("go"),
+		dependencyAnalyzer:            dependencyAnalyzer.AnalyzerFactory("go", ctx),
```

Comment:
the default should not be go here. It needs to be detected or taken as an input (as a temporary workaround)

Reply:
Have moved this to command line argument.